### PR TITLE
Add debug endpoint to buildtracker to get builds

### PR DESCRIPTION
**Pull request by burmudar**
This adds a `/-/debug/{buildNumber}` endpoint to buildtracker

The endpoint is "protected" with Basic Auth. The credentials can be found in the dev-experience vault on 1password
## Test plan
Unit tests and local manual testing

### CODE REVIEW

File: dev/build-tracker/main.go [hunk 1]
```diff
@@ -55,18 +61,90 @@ func configFromEnv() (*config, error) {
...
 
+	err = envVar("BUILDTRACKER_PRODUCTION", &c.Production)
+	if err != nil {
+		c.Production = false
+	}
+
+	if c.Production {
+		_ = envVar("BUILDTRACKER_DEBUG_PASSWORD", &c.DebugPassword)
+		if c.DebugPassword == "" {
+			return nil, errors.New("BT_DEBUG_PASSWORD is required when BT_PRODUCTION is true")
```
Comment (burmudar) [inline dev/build-tracker/main.go:72]:
```suggestion
			return nil, errors.New("BUILDTRACKER_DEBUG_PASSWORD is required when BUILDTRACKER_PRODUCTION is true")
```

File: dev/build-tracker/util.go [hunk 2]
```diff
@@ -22,12 +23,27 @@ func intp(v *int) int {
...
 
-func envVar(name string, target *string) error {
+func envVar[T any](name string, target *T) error {
 	value, exists := os.LookupEnv(name)
 	if !exists {
 		return errors.Newf("%s not found in environment", name)
 	}
 
-	*target = value
+	switch p := any(target).(type) {
```
Comment (bobheadxi) [inline dev/build-tracker/util.go:32]: 👁️

File: dev/build-tracker/util.go [hunk 3]
```diff
@@ -22,12 +23,27 @@ func intp(v *int) int {
...
+				return err
+			}
+
+			*p = v
+		}
+	case *string:
+		{
+			*p = value
+		}
+	}
```
Comment (bobheadxi) [inline dev/build-tracker/util.go:46]: Should we have a default panic case?

Comment (burmudar) [reply dev/build-tracker/util.go:46]: yes! Good catch!

**Review by bobheadxi — APPROVED** [review]
nice!

**Pull request merged**